src/GafferScene/RenderController.cpp

```diff
@@ -316,6 +330,35 @@ class RenderController::SceneGraph
 						m_changedComponents |= AttributesComponent;
 					}
 				}
+
+				if( ( m_changedComponents & AttributesComponent ) || ( changedGlobals & ( TransformBlurGlobalComponent | DeformationBlurGlobalComponent ) ) )
```

Comment:
The redundant work thing I have fixed.  Lifting out of the root I haven't done yet - it will be easy to do, but I just want to understand why - these settings aren't inherited ( the inheritance uses m_fullAttributes ), they're only used for setting transforms or objects at the current location, and I don't think it's valid to do that at the root?  Maybe we want to possibly allow for transforms at the root in the future?

Reply:
I was thinking this was necessary in order to correctly inherit changes to blur settings from the global attributes. But I now see this inheritance is already taken care of by the `parentAttributesChanged` flag, which will cause us to do `m_changedComponents |= AttributesComponent` and enter the code via the path you've already got. I've tested this interactively too, and all is well.